addons/mail/static/src/core/common/model/make_store.js

```diff
@@ -0,0 +1,314 @@
+/* @odoo-module */
+
+import { reactive } from "@odoo/owl";
+import {
+    IS_FIELD_DEFINITION_SYM,
+    IS_RECORD_FIELD_SYM,
+    IS_RECORD_SYM,
+    STORE_SYM,
+    modelRegistry,
+    _0,
+    isField,
+    isMany,
+    isRelation,
+} from "./misc";
+import { Store } from "./store";
+import { onChange } from "@mail/utils/common/misc";
+import { RecordList } from "./record_list";
+import { Record } from "./record";
+
+export function makeStore(env) {
+    const recordByLocalId = reactive(new Map());
+    const dummyStore = new Store();
+    dummyStore.recordByLocalId = recordByLocalId;
+    dummyStore.env = env;
+    const _ = { store: dummyStore };
+    Record.store0 = dummyStore;
+    const Models = {};
+    for (const [name, _OgClass] of modelRegistry.getEntries()) {
+        /** @type {typeof Record} */
+        const OgClass = _OgClass;
+        if (_.store[name]) {
+            throw new Error(`There must be no duplicated Model Names (duplicate found: ${name})`);
+        }
+        // classes cannot be made reactive because they are functions and they are not supported.
+        // work-around: make an object whose prototype is the class, so that static props become
+        // instance props.
+        /** @type {typeof Record} */
+        const Model = Object.create(OgClass);
+        // Produce another class with changed prototype, so that there are automatic get/set on relational fields
+        const Class = {
+            [OgClass.name]: class extends OgClass {
+                [IS_RECORD_SYM] = true;
+                constructor() {
+                    super();
+                    const this0 = this;
```

Comment:
`this` depends on context and I find it much harder to read/understand/grep than using the full word of the object. For me we should ban this keyword as much as possible.

If you need to gain lines length, I'd rather we split the code in many small methods instead to reduce the insane indentation level we have currently.